src/languageservice/decompressProvider.ts

```diff
@@ -7,18 +7,59 @@
 
 import {IDecompressProvider, IPackage} from './interfaces';
 import  {ILogger} from '../models/interfaces';
-const decompress = require('decompress');
+import { PlatformInformation } from '../models/platform';
+
+const DecompressZip = require('decompress-zip');
+const DecompressTar = require('tar');
 
 export default class DecompressProvider implements IDecompressProvider {
-    public decompress(pkg: IPackage, logger: ILogger): Promise<void> {
+
+    private _isWindows: boolean;
+
+    constructor() {
+        PlatformInformation.getCurrent().then((platform: PlatformInformation) => {
+            this._isWindows = platform.isWindows();
+        });
+    }
+
+    private decompressZip(pkg: IPackage, logger: ILogger): Promise<void> {
+        const unzipper = new DecompressZip(pkg.tmpFile.name);
         return new Promise<void>((resolve, reject) => {
-            decompress(pkg.tmpFile.name, pkg.installPath).then(files => {
-                    logger.appendLine(`Done! ${files.length} files unpacked.\n`);
-                    resolve();
-                }).catch(decompressErr => {
-                        logger.appendLine(`[ERROR] ${decompressErr}`);
-                        reject(decompressErr);
-                });
+            let totalFiles = 0;
+            unzipper.on('progress', async (index, fileCount) => {
+                totalFiles = fileCount;
+            });
+            unzipper.on('extract', async () => {
+                logger.appendLine(`Done! ${totalFiles} files unpacked.\n`);
+                resolve();
+            });
+            unzipper.on('error', async (decompressErr) => {
+                logger.appendLine(`[ERROR] ${decompressErr}`);
+                reject(decompressErr);
+            });
+            unzipper.extract({ path: pkg.installPath });
         });
     }
+
+    private decompressTar(pkg: IPackage, logger: ILogger): Promise<void> {
+        let totalFiles = 0;
+        return DecompressTar.extract({
+            file: pkg.tmpFile.name,
+            cwd: pkg.installPath,
+            onentry: () => { totalFiles++; },
+            onwarn: (warn) => {
+                if (warn.data && !warn.data.recoverable) {
+                    logger.appendLine(`[ERROR] ${warn.message}`);
+                }
+            }
+        }, () => { logger.appendLine(`Done! ${totalFiles} files unpacked.\n`); });
+    }
+
+    public decompress(pkg: IPackage, logger: ILogger): Promise<void> {
+        if (this._isWindows) {
```

Comment:
Why are you relying on the platform for determining what type of decompression to do? Looking at the file extension instead is the usual way of doing this.

Reply:
We cannot use the file extension here because it's a temporary file with a `.tmp` extension

Reply:
I would suggest doing what I did for service-downloader and instead of using .tmp use the same file extension as the downloaded file (I assume the logic is the same here). It just makes more sense to do that anyways - there's no good reason we'd want to rename it to .tmp.

Reply:
(and the main reason I'm asking is because relying on the platform is making a race condition possible here if this function is called before the platform query returns)